File: Reuniao.java
public class Reuniao extends Compromisso{
    private boolean adiavel;
    private String assunto;

    public Reuniao(int id, Data data, Hora hora, int duracao, int prioridade, boolean adiavel, String assunto){
        super(id, 'R', data, hora, duracao, prioridade, assunto);
        this.adiavel = adiavel;
        this.assunto = assunto;
        calcularPrioridadeFinal();
    }

    @Override
    public void calcularPrioridadeFinal(){
        //multiplicador 4x
        int multiplicador = 4;
        //se n adiavel  +1
        if(!adiavel){
            multiplicador += 1;
        }
        this.prioridadeFinal = this.prioridade * multiplicador;
    }

    @Override
    public String getDescricao(){
        return "Reunião de Departamento";
    }

    @Override
    public boolean isAdiavel(){
        return adiavel;
    }

    public String getAssunto(){
        return assunto;
    }
}

File: python/reuniao.py
from compromisso import Compromisso

class Reuniao(Compromisso):
    def __init__(self, id, data, hora, duracao, prioridade, adiavel, assunto):
        super().__init__(id, 'R', data, hora, duracao, prioridade, assunto)
        self.adiavel = adiavel
        self.assunto = assunto
        self.calcularPrioridadeFinal()
    
    def calcularPrioridadeFinal(self):
        multiplicador = 4
        if not self.adiavel:
            multiplicador += 1
        self.prioridadeFinal = self.prioridade * multiplicador
    
    def getDescricao(self):
        return "Reunião de Departamento"
    
    def isAdiavel(self):
        return self.adiavel
    
    def getAssunto(self):
        return self.assunto